"Internal Server Error" for proxied images in hosted instance

When I read an article, I currently get a "broken image" icon instead of any picture. The url is `https://reader.miniflux.app/proxy/some-long-token`, and instead of an image it simply returns Internal Server Error. I can't say exactly when it started, a couple of days ago maybe?

There was a mention of some security issue with the image proxy in recent release notes. Could that be related perhaps?

Seems any feed (old or new in v2.0.43) with proxyed images has this issue. 
Error log like:
```
[ERROR] [Proxy] Unable to initialize HTTP client: Get "/proxy/UG9ryouUC7_LiHUeOjYqV8A3qu__VCSuj1YVlud1JOA=/aHR0cHM6Ly9jZG4uYmVla2thLmNvbS9ibG9naW1nL2Fzc2V0LzIwMjMwMS9iZzIwMjMwMTE3MDgud2VicA==": unsupported protocol scheme ""
```

> Seems any feed (old or new in v2.0.43) with proxyed images has this issue. Error log like:
> 
> ```
> [ERROR] [Proxy] Unable to initialize HTTP client: Get "/proxy/UG9ryouUC7_LiHUeOjYqV8A3qu__VCSuj1YVlud1JOA=/aHR0cHM6Ly9jZG4uYmVla2thLmNvbS9ibG9naW1nL2Fzc2V0LzIwMjMwMS9iZzIwMjMwMTE3MDgud2VicA==": unsupported protocol scheme ""
> ```

After change conf `PROXY_IMAGES` to `PROXY_OPTION` works fine now...

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -1226,6 +1226,66 @@ func TestProxyMediaTypes(t *testing.T) {
 	}
 }
 
+func TestProxyMediaTypesWithDuplicatedValues(t *testing.T) {
+	os.Clearenv()
+	os.Setenv("PROXY_MEDIA_TYPES", "image,audio, image")
+
+	parser := NewParser()
+	opts, err := parser.ParseEnvironmentVariables()
+	if err != nil {
+		t.Fatalf(`Parsing failure: %v`, err)
+	}
+
+	expected := []string{"audio", "image"}
+	if len(expected) != len(opts.ProxyMediaTypes()) {
+		t.Fatalf(`Unexpected PROXY_MEDIA_TYPES value, got %v instead of %v`, opts.ProxyMediaTypes(), expected)
+	}
+
+	resultMap := make(map[string]bool)
+	for _, mediaType := range opts.ProxyMediaTypes() {
+		resultMap[mediaType] = true
+	}
+
+	for _, mediaType := range expected {
+		if !resultMap[mediaType] {
+			t.Fatalf(`Unexpected PROXY_MEDIA_TYPES value, got %v instead of %v`, opts.ProxyMediaTypes(), expected)
+		}
+	}
+}
+
+func TestProxyImagesOptionBackwardCompatibility(t *testing.T) {
+	os.Clearenv()
+	os.Setenv("PROXY_IMAGES", "all")
+
+	parser := NewParser()
+	opts, err := parser.ParseEnvironmentVariables()
+	if err != nil {
+		t.Fatalf(`Parsing failure: %v`, err)
+	}
+
+	expected := []string{"image"}
+	if len(expected) != len(opts.ProxyMediaTypes()) {
+		t.Fatalf(`Unexpected PROXY_MEDIA_TYPES value, got %v instead of %v`, opts.ProxyMediaTypes(), expected)
+	}
+
+	resultMap := make(map[string]bool)
+	for _, mediaType := range opts.ProxyMediaTypes() {
+		resultMap[mediaType] = true
+	}
+
+	for _, mediaType := range expected {
+		if !resultMap[mediaType] {
+			t.Fatalf(`Unexpected PROXY_MEDIA_TYPES value, got %v instead of %v`, opts.ProxyMediaTypes(), expected)
+		}
+	}
+
+	expectedProxyOption := "all"
+	result := opts.ProxyOption()
+	if result != expectedProxyOption {
+		t.Fatalf(`Unexpected PROXY_OPTION value, got %q instead of %q`, result, expectedProxyOption)
+	}
+}
+
 func TestDefaultProxyMediaTypes(t *testing.T) {
 	os.Clearenv()
 

--- a/config/parser.go
+++ b/config/parser.go
@@ -141,7 +141,6 @@ func (p *Parser) parseLines(lines []string) (err error) {
 		// kept for compatibility purpose
 		case "PROXY_IMAGES":
 			p.opts.proxyOption = parseString(value, defaultProxyOption)
-			p.opts.proxyMediaTypes = append(p.opts.proxyMediaTypes, "image")
 		case "PROXY_HTTP_CLIENT_TIMEOUT":
 			p.opts.proxyHTTPClientTimeout = parseInt(value, defaultProxyHTTPClientTimeout)
 		case "PROXY_OPTION":
@@ -297,9 +296,16 @@ func parseStringList(value string, fallback []string) []string {
 	}
 
 	var strList []string
+	strMap := make(map[string]bool)
+
 	items := strings.Split(value, ",")
 	for _, item := range items {
-		strList = append(strList, strings.TrimSpace(item))
+		itemValue := strings.TrimSpace(item)
+
+		if _, found := strMap[itemValue]; !found {
+			strMap[itemValue] = true
+			strList = append(strList, itemValue)
+		}
 	}
 
 	return strList

